File: mw/getdistmat_euclidean.m
function [distmat] = getdistmat_euclidean(s)
% Get matrix of distances from location matrix
% 
n = size(s,1);
d = size(s,2);
dist = zeros(n,n);
for j = 1:d;
 dist = dist + (repmat(s(:,j),1,n)-repmat(s(:,j)',n,1)).^2;
end;
distmat = sqrt(dist);
    
end

File: R/get_distmat_euclidean.R
#' Calculate Euclidean Distance Matrix
#'
#' Computes the matrix of Euclidean distances between all pairs of locations
#' in the input matrix.
#'
#' @param s A numeric matrix where rows represent locations and columns represent dimensions.
#' @return A numeric matrix of pairwise Euclidean distances between locations.
get_distmat_euclidean <- function(s) {
  # Use R's built-in dist function and convert to matrix
  distmat <- as.matrix(stats::dist(s, method = "euclidean"))
  
  return(distmat)
}

#' @rdname get_distmat_euclidean
getdistmat_euclidian <- function(s) {
  get_distmat_euclidean(s = s)
}
